dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```
@@ -0,0 +1,638 @@
+
+    private static void writeMap(DataOutputStream dos, MapIterable map) throws IOException {
+        dos.writeInt(map.size());
+        for (Map.Entry<String, Object> entry : map) {
+            dos.writeUTF(entry.getKey());
+            Object obj = entry.getValue();
+            if (obj instanceof Collection) {
+                dos.writeByte(COLLECTION);
+                Collection<?> data = (Collection<?>) obj;
+                dos.writeInt(data.size());
+                boolean writeType = true;
+                for (Object colObj : data) {
+                    // write primative.
+                    if (writeType) {
+                        writePrimitiveType(dos, colObj.getClass());
+                        writeType = false;
+                    }
+                    writePrimitive(dos, colObj);
+                }
+            } else if (obj instanceof Map) {
+                dos.writeByte(MAP);
+                writeMap(dos, toMapOrDictionary(obj));
+            } else if (obj.getClass().isArray()) {
+                dos.writeByte(ARRAY);
+                Class<?> arrayType = obj.getClass();
+                arrayType = arrayType.getComponentType();
+                writePrimitiveType(dos, arrayType);
+                dos.writeBoolean(arrayType.isPrimitive());
+                int len = Array.getLength(obj);
+                dos.writeInt(len);
+                for (int i = 0; i < len; i++) {
+                    writePrimitive(dos, Array.get(obj, i));
+                }
+            } else {
+                writePrimitiveType(dos, obj.getClass());
+                writePrimitive(dos, obj);
+            }
+        }
+    }
+
+    private static MapIterable toMapOrDictionary(Object obj) {
+        if (obj instanceof Dictionary) {
+            return new DictionaryMapIterableImpl((Dictionary<String, Object>) obj);
+        } else if (obj instanceof Map) {
+            return new MapIterableImpl((Map<String, Object>) obj);
+        }
+        return null;
```

throw IllegalArgumentException instead of returning null.